cmd/2019_7/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"flag"
+
+	"github.com/afarbos/aoc/pkg/intcode"
+	"github.com/afarbos/aoc/pkg/mathematic"
+	"github.com/afarbos/aoc/pkg/perm"
+	"github.com/afarbos/aoc/pkg/read"
+	"github.com/afarbos/aoc/pkg/utils"
+)
+
+var flagInput string
+
+func init() {
+	utils.Init(&flagInput)
+}
+
+func compute(instructions []int, inputs []chan int, amplifier, nextAmplifier int, done chan struct{}) {
+	localInstructions := make([]int, len(instructions))
+	copy(localInstructions, instructions)
+	intcode.Compute(localInstructions, intcode.Equals, inputs[amplifier], inputs[nextAmplifier])
+	done <- struct{}{}
+}
+
+func findMaxThrusterSignalWithSettings(instructions []int, settingsSet map[int]struct{}) int {
+	var signals []int
+
+	perm.Ints(settingsSet, func(settings []int) {
+		var (
+			inputs = make([]chan int, 0, len(settings))
+			done   = make(chan struct{})
+		)
+
+		for range settings {
+			inputs = append(inputs, make(chan int, 1))
+		}
+
+		for amplifier, phase := range settings {
+			go compute(instructions, inputs, amplifier, (amplifier+1)%len(settings), done)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '1', create a named constant for it](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)